rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/FileMetadata.java

```diff
@@ -41,30 +48,68 @@
   private long size;
   private long lastModified;
   private long currentFileSize;
-
   private boolean needsRefresh = true;
+  private int generation;
```

Comment:
rename it to `generationNumber` as we have used this term everywhere for consistency.

Reply:
It can be initialized to '-1' to denote an invalid generation instead of. taking 0 as default.